src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -1433,3 +1513,72 @@ def merge_metadata(a_metadata, b_metadata, *, source, update=(), min_=(), max_=(
         result_directory=r_a.result_directory,
         app_version_info=app_version_info,
     )
+
+
+@contextmanager
+def open_global_sensitivity_analysis_result_file(
+    results_directory: Path,
+) -> Optional[Iterator[h5py.File]]:
+    """
+    Open a global sensitivity analysis result file.
+    """
+    filename = results_directory / "uq_result"
+    ignored_file = results_directory / "uq_result.creating"
+    # Avoid to read result files with incomplete metadata.
+    if not ignored_file.is_file():
+        file = _open_result_file(filename)
+        yield file
+        file.close()
```

Comment:
Always surround with try/finally, otherwise the code after the `yield` will not be reached in the case of an exception:

```suggestion
        file = _open_result_file(filename)
        try:
            yield file
        finally:    
            file.close()
```

However, `h5py.File` supports the context manager, so this is even better:

```suggestion
        with _open_result_file(filename):
            yield file    
```